### DISCUSSION TITLE
Tensorrt inference on Yolov8 -OBB (Oriented Bounding Box)

### DISCUSSION THREAD

**Discussion by kkamalrajk** [ultralytics/ultralytics#8071] [Q&A]
👋 Hello,

I successfully converted the 'yolov8x.pt' detection model to .engine format, experiencing a remarkable 2x improvement in inference speed on V100 GPUs.

However, when attempting the same with the 'yolov8x-obb.pt' (Oriented Bounding Box) model, the achieved speed boost is only 1.09x on the identical V100 16GB GPU. My expectation was to attain at least a 2x improvement in inference speed with the TensorRT version.

I am seeking guidance on potential optimizations or strategies to enhance the performance and meet the anticipated 2x speed boost on the oriented bounding box model.

Appreciate your assistance!

Regards,
Kamal Raj

**Comment by pderrenger** [comment]
@kkamalrajk 👋 Hi Kamal Raj,

Great to hear about your success with the 'yolov8x.pt' model and TensorRT! 🚀

For the 'yolov8x-obb.pt' model, achieving the same level of speedup can be more challenging due to the additional complexity of predicting oriented bounding boxes. However, here are a few suggestions that might help you optimize the performance further:

1. **Precision Tuning**: Ensure that you're using the appropriate precision (FP16 or INT8) during the TensorRT conversion, which can provide significant speed improvements.

2. **Layer Fusion**: Verify that TensorRT is fusing layers optimally. Sometimes manual intervention is required to ensure that the graph is as efficient as possible.

3. **Batch Size**: Experiment with different batch sizes to find the sweet spot for your GPU. Larger batch sizes can sometimes improve throughput.

If you've already tried these and are still not seeing the expected improvements, it might be worth reaching out to the NVIDIA forums or support channels for more hardware-specific guidance.

Keep pushing the limits! 💪

**Reply by kkamalrajk** [reply]
@pderrenger 👋 Hi Paula Derrenger,

I used the code below to convert the OBB model to the .engine format:

'''
from ultralytics import YOLO
model = YOLO('yolov8x-obb.pt')
model.export(format='engine')
'''

Could you confirm if this code incorporates the optimisations you mentioned? If not, could your team at Ultralytics modify the 'export' function to accommodate OBB models? This adjustment could benefit others using OBB models as well.

Regards,
Kamal Raj

**Reply by pderrenger** [reply]
@kkamalrajk Hi Kamal Raj,

The code snippet you've provided does indeed convert the model to TensorRT `.engine` format, but it doesn't explicitly control for all potential optimizations. The `export` function automates the conversion process with general optimizations. For OBB models, which have unique requirements, additional manual tuning might be necessary to achieve the best performance.

While we continuously work on improving our export functionalities, incorporating specific optimizations for OBB models is a great suggestion and something we can consider for future updates. In the meantime, you might want to explore manual TensorRT optimizations based on your model's specific needs.

Keep an eye on our updates, and thanks for your input! 🌟
